### PR TITLE
Increase unit tests coverage in package api

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: This PR increases coverage of pkg/api/types.go from `40.32%` to `100%`.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: helps solve #2576  

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [x] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -769,11 +768,6 @@ func (a *AgentPoolProfile) HasDisks() bool {
 	return len(a.DiskSizesGB) > 0
 }
 
-// IsAcceleratedNetworkingEnabled returns true if the customer enabled Accelerated Networking
-func (a *AgentPoolProfile) IsAcceleratedNetworkingEnabled() bool {
```
Comment (CecileRobertMichon) [inline pkg/api/types.go]: I was going to write unit tests for this but then I realized there's no point in having this function so I removed it

File: pkg/api/types.go [hunk 2]
```diff
@@ -661,7 +660,7 @@ func (p *Properties) HasStorageAccountDisks() bool {
 			return true
 		}
 	}
-	if p.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision() && p.OrchestratorProfile.KubernetesConfig.PrivateCluster.JumpboxProfile.StorageProfile == StorageAccount {
+	if p.OrchestratorProfile != nil && p.OrchestratorProfile.KubernetesConfig != nil && p.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision() && p.OrchestratorProfile.KubernetesConfig.PrivateCluster.JumpboxProfile.StorageProfile == StorageAccount {
```
Comment (CecileRobertMichon) [inline pkg/api/types.go]: stricter nil checks

File: pkg/api/types.go [hunk 3]
```diff
@@ -642,15 +641,15 @@ func (p *Properties) HasManagedDisks() bool {
...
+	if p.OrchestratorProfile != nil && p.OrchestratorProfile.KubernetesConfig != nil && p.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision() && p.OrchestratorProfile.KubernetesConfig.PrivateCluster.JumpboxProfile.StorageProfile == ManagedDisks {
 		return true
 	}
 	return false
 }
 
 // HasStorageAccountDisks returns true if the cluster contains Storage Account Disks
 func (p *Properties) HasStorageAccountDisks() bool {
-	if p.OrchestratorProfile.OrchestratorType == OpenShift {
+	if p.OrchestratorProfile != nil && p.OrchestratorProfile.OrchestratorType == OpenShift {
```
Comment (CecileRobertMichon) [inline pkg/api/types.go]: stricter nil checks

File: pkg/api/types.go [hunk 4]
```diff
@@ -642,15 +641,15 @@ func (p *Properties) HasManagedDisks() bool {
 			return true
 		}
 	}
-	if p.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision() && p.OrchestratorProfile.KubernetesConfig.PrivateCluster.JumpboxProfile.StorageProfile == ManagedDisks {
+	if p.OrchestratorProfile != nil && p.OrchestratorProfile.KubernetesConfig != nil && p.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision() && p.OrchestratorProfile.KubernetesConfig.PrivateCluster.JumpboxProfile.StorageProfile == ManagedDisks {
```
Comment (CecileRobertMichon) [inline pkg/api/types.go]: stricter nil check

File: pkg/api/types.go [hunk 5]
```diff
@@ -443,13 +443,12 @@ type AgentPoolProfile struct {
 	Distro                       Distro               `json:"distro,omitempty"`
 	Role                         AgentPoolProfileRole `json:"role,omitempty"`
 	AcceleratedNetworkingEnabled bool                 `json:"acceleratedNetworkingEnabled,omitempty"`
-
```
Comment (CecileRobertMichon) [inline pkg/api/types.go]: removed space

File: parts/k8s/kubernetesagentresourcesvmss.t [hunk 6]
```diff
@@ -53,7 +53,7 @@
               "name": "[variables('{{.Name}}VMNamePrefix')]",
               "properties": {
                 "primary": true,
-                "enableAcceleratedNetworking" : "{{.IsAcceleratedNetworkingEnabled}}",
+                "enableAcceleratedNetworking" : "{{.AcceleratedNetworkingEnabled}}",
```
Comment (CecileRobertMichon) [inline parts/k8s/kubernetesagentresourcesvmss.t]: See comment below, the function was superfluous, we can just check the bool type instead

File: pkg/api/vlabs/types.go [hunk 7]
```diff
@@ -609,11 +609,6 @@ func (a *AgentPoolProfile) SetSubnet(subnet string) {
 	a.subnet = subnet
 }
 
-// IsAcceleratedNetworkingEnabled returns true if the customer enabled Accelerated Networking
-func (a *AgentPoolProfile) IsAcceleratedNetworkingEnabled() bool {
```
Comment (CecileRobertMichon) [inline pkg/api/vlabs/types.go]: same as above

File: parts/k8s/kubernetesagentresourcesvmas.t [hunk 8]
```diff
@@ -1,5 +1,5 @@
     {
-{{if .IsAcceleratedNetworkingEnabled}}
+{{if .AcceleratedNetworkingEnabled}}
```
Comment (CecileRobertMichon) [inline parts/k8s/kubernetesagentresourcesvmas.t]: See comment below, the function was superfluous, we can just check the bool type instead

**Comment by CecileRobertMichon** [comment]
tested accelerated networking still works at https://jenkins.azure-containers.io/view/acs-engine/job/k8s-accelerated-networking/186/console

**Comment by jackfrancis** [comment]
/approve
/lgtm

🎉

**Pull request merged**